web/src/settings_components.ts

```diff
@@ -1016,6 +1035,28 @@ export function populate_data_for_realm_settings_request(
                     });
                     continue;
                 }
+
+                if (
+                    property_name === "allow_resolve_topic" ||
+                    property_name === "can_resolve_topics_group"
+                ) {
+                    let new_user_group_id: number;
+                    if (input_value === false) {
+                        const new_user_group = user_groups.get_user_group_from_name("role:nobody");
+                        assert(new_user_group !== undefined);
+                        new_user_group_id = new_user_group?.id;
+                    } else {
+                        assert(can_resolve_topics_group_widget !== null);
+                        const can_resolve_topics_group_id = can_resolve_topics_group_widget.value();
+                        assert(typeof can_resolve_topics_group_id === "number");
+                        new_user_group_id = can_resolve_topics_group_id;
+                    }
+                    data.can_resolve_topics_group = JSON.stringify({
+                        new: new_user_group_id,
+                        old: realm.realm_can_resolve_topics_group,
+                    });
```

Comment:
I wonder if we can do something simple here. Using two elements for one setting field makes it somewhat non-trivial. I will think more if we can do something and will post here if I can.